hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1363,6 +1363,10 @@ public void close() {
     // before this point
     this.index.close();
     this.tableServiceClient.close();
+    // shutdown metrics
+    if (this.metrics != null && this.metrics.getMetrics() != null) {
```

Comment:
hey @prashantwason : can you review this. we have few tests written w/ metadata table. after this fix those tests are failing since we shutdown the metrics, the metrics from registry are not accessible once the write operation is complete. 
Test of interest:
https://github.com/apache/hudi/blob/f72bf39e5e0f7431a8772961af988da346dfb376/hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieBackedMetadata.java#L2902

Reply:
I could not figure out a way to test the metrics otherwise. but the fix also makes sense i.e. closing the metrics when we close the write client. 
let us know if you have any pointers.

Reply:
Does this imply the metrics will not be available to scrape after the client closes? That can be a problem if you have a longer scrape interval like 30 seconds for example